### PR TITLE
feat(wandb): improve callback

### PR DESCRIPTION
**Pull request by borisdayma**
Following changes are implemented:

* ensure callback is executed at the end -> we would get errors when the callback was executed before `smooth_loss` was defined
* `time` is not logged anymore as it is formatted as hh:mm which cannot be plotted (and create an error on associated graph). Wall time and relative time are already automatically logged by wandb
* we keep track of cumulative epoch -> this allows to handle the cases where the validation set changes between training stages (as `n_iter` could vary)

The changes allow to call `learner(…, cbs=WandbCallback())` and multiple times `.fit` methods (even changing validation set) and add data to the same graphs as in [this run](https://app.wandb.ai/borisd13/fastai2/runs/jmyeskcb?workspace=user-borisd13).

### CODE REVIEW

**Comment by borisdayma** [comment]
Remaining TODO:

* prevent logging during `lr_find` such as [this run](https://app.wandb.ai/borisd13/fastai2/runs/1vnjip7c?workspace=user-borisd13) -> could we update the `no_logging` function by adding a `no_logging` attribute so it could be used in the callback?
* add tests -> I can look into this when we finalize base functionality

Possible improvements:

* log data such as confusion matrix and top losses samples (depending on data class type) after each epoch -> dataframes are nice to visualize like in [this example](https://app.wandb.ai/lavanya/building-neural-nets/runs/4d2pfleq?workspace=user-lavanya)

**Comment by borisdayma** [comment]
Slightly confused about the build error, when I run diff_nb_script I see no output.

File: dev/70_callback_wandb.ipynb [hunk 1]
```diff
@@ -74,12 +74,14 @@
     "#export\n",
```
Comment (sgugger) [inline dev/70_callback_wandb.ipynb]: <p><code>self.epoch</code> normally points to <code>self.learn.epoch</code> and is the integer of the current epoch. I think it's dangerous to use the same name for something slightly different. Maybe make it a private variable?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastai_dev/pull/292/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (borisdayma) [reply dev/70_callback_wandb.ipynb]: <p>You're right. Also at the end of one epoch <code>self.learn.epoch = 0</code> while we want <code>epoch=1</code> (because training losses will have been logged) or even more if we had a few phases before.</p><p>What do you think of starting all variables with <code>_wandb</code> to make it completely clear and safe: <code>_wandb_watch_called</code>, <code>_wandb_step</code>, <code>_wandb_epoch</code></p>

Comment (sgugger) [reply dev/70_callback_wandb.ipynb]: That sounds good.

Comment (borisdayma) [reply dev/70_callback_wandb.ipynb]: Variables changed and [test run](https://app.wandb.ai/borisd13/fastai2/runs/kk938yrf?workspace=user-borisd13) successful.

Last issue is related to preventing logging during `lr_find`. Any suggestion?

Also an idea is to add `Interpretation.top_losses()` as a summary log at the end of each epoch or at the end of training (with potentially confusing matrix for classification) but we can keep that for later.

Comment (sgugger) [reply dev/70_callback_wandb.ipynb]: Running `Learner.lr_find()` adds a callback `LRFinder` which then is added as an attribute of the `Learner`. I think simply testing `hasattr(learn, 'lr_finder')` should be enough to detect if we are in a `lr_find` and not log anything.

Comment (borisdayma) [reply dev/70_callback_wandb.ipynb]: Thanks it worked great!

**Comment by borisdayma** [comment]
I took advantage to add alternative use in the description: `learn = learner(..., cbs=WandbCallback())`

Feel free to remove `entity` from the docs after testing as it should not be needed anymore.

**Comment by sgugger** [comment]
Thanks!
